### PR TITLE
[ci] Package C-API in nightly builds

### PR DESCRIPTION
**Pull request by feisuzhu**
Issue: #

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -639,6 +639,12 @@ jobs:
             registry.taichigraphics.com/taichidev-androidsdk18.04:v0.0.6 \
             /home/dev/taichi/.github/workflows/scripts/aot-demo.sh smoke-test-unity-demo
 
+      - name: Save C-API shared object (Android)
+        uses: actions/upload-artifact@v3
+        with:
+          path: build/libtaichi_c_api.so
```
Comment (ailzhang) [inline .github/workflows/testing.yml:645]: @feisuzhu OOC is there a fixed URL that other CI can download from or do they need to find out the link manually? ;)

Comment (feisuzhu) [reply .github/workflows/testing.yml:645]: No, there's no fixed url :(

**Review by ailzhang — APPROVED** [review]
Awesome!! cc: @imcom @YuCrazing Once this PR is in you can get c_api.so in the nightly taichi wheel ;)

**Comment by imcom** [comment]
So the installation of the wheel puts the .so somewhere on the system ?

**Comment by ailzhang** [comment]
@imcom Yup it'll be available somewhere in your python package once you `pip install taichi-nightly` ;)

**Pull request merged**